File: 1945/solution.java
class Solution {
    public int getLucky(String s, int k) {
        int total = 0;
        for (int i = 0; i < s.length(); i++) {
            int num = s.charAt(i) - 'a' + 1;
            while (num > 0) {
                total += num % 10;
                num /= 10;
            }
        }
        k--;
        while (k > 0) {
            int newTotal = 0;
            while (total > 0) {
                newTotal += total % 10;
                total /= 10;
            }
            total = newTotal;
            k--;
        }
        return total;
    }
}
// time O(n * k)
// space O(1)

File: 1945/solution.go
package main

func getLucky(s string, k int) int {
	total := 0
	for _, v := range s {
		num := int(v - 'a' + 1)
		for num > 0 {
			total += num % 10
			num /= 10
		}
	}
	k--
	for k > 0 {
		newTotal := 0
		for total > 0 {
			newTotal += total % 10
			total /= 10
		}
		total = newTotal
		k--
	}
	return total
}